fuel-vm/src/interpreter/memory.rs

```diff
@@ -202,36 +202,13 @@ where
     ///
     /// Will panic if data overlaps with `addr[..|data|[`
     pub(crate) fn try_mem_write(&mut self, addr: usize, data: &[u8]) -> Result<(), RuntimeError> {
-        let ax = addr.checked_add(data.len()).ok_or(PanicReason::ArithmeticOverflow)?;
-
-        let range = (ax <= VM_MAX_RAM as usize)
-            .then(|| MemoryRange::new(addr as Word, 32))
-            .ok_or(PanicReason::MemoryOverflow)?;
-
-        self.has_ownership_range(&range)
-            .then(|| {
-                let src = data.as_ptr();
-                let dst = &mut self.memory[addr] as *mut u8;
-
-                unsafe {
-                    ptr::copy_nonoverlapping(src, dst, data.len());
-                }
-            })
-            .ok_or_else(|| PanicReason::MemoryOwnership.into())
+        let registers = OwnershipRegisters::new(self);
```

Comment:
[nit] Hmm, maybe we can add one helper method like `fn registers(&self)` and use it `try_mem_write(addr, data, self.registers(), &mut self.memory)`